### ISSUE TITLE
Move away from percy.

### ISSUE THREAD

**Issue by Pomax** [mozilla/foundation.mozilla.org#8977]
Based on chats with engineers and designers, only engineers use Percy's visual regression to verify PRs don't introduce visual regressions. Given its cost, and the delay it puts on PRs, I would like to recommend we stop using Percy integration, and instead make visual regression testing a local testing step that engineers can run prior to pushing their code, as well as when checking out someone else's branch for testing, as a new `inv test-visual` that can be run on its own, as well as running as part of the general `inv test` supertask.

We have both Cypress and Playwright available for this purpose, so I would recommend we start by removing the percy "plugin" for cypress and use the built-in screenshot functions to save images locally rather than sending DOM snapshots to browserstack. BS does have a guide up that explains how to set up cypress for local visual regression testing on https://www.browserstack.com/guide/visual-regression-testing-with-cypress that we can use to get things up and running.

Ideally, this ultimately happens as Github Action that we can enable again, we had already undertaken some of that work in the past but the diffing solution (based on playwright, using chrome and firefox at several breakpoints) was never quite finished because we never dedicated enough time to it in order to cross the t's and dot the i's - it's close (see https://github.com/MozillaFoundation/ci-image-diff), but someone will probably have to (help) clean up the last outstanding issues on that.

**Comment by tbrlpld** [comment]
I'd be hesitant to first remove Percy before we have a replacement, especially with the ongoing CSS transition to Tailwind.

**Comment by Pomax** [comment]
The main problem is that Percy costs _a lot_ (like, _a lot_ a lot), and we go over our current plan's number of screenshots after 10 days into a month, paying overage charges for the rest of the month. Making this part of the dev workflow, rather than CI, doesn't feel like a big change: you (hopefully! =D) run `inv test` already anyway, making that perform visual regression testing for your branch code should pretty much work just as well as running it remotely on push.

**Comment by tbrlpld** [comment]
I wonder if there is something in the config we can do to make the existing plan last a bit longer. E.g. don't run it on "draft" PRs etc.

And yea, I am not married to Percy itself. If we have another CI tool that offers similar functionality, that would totally work.

**Comment by Pomax** [comment]
Hence https://www.browserstack.com/guide/visual-regression-testing-with-cypress - we already use cypress, so having someone sit down and turn that into a new github workflow action would probably be good enough to (once that works) turn off Percy.

**Comment by mofodevops** [comment]
## Team Review of Percy Usage
Web Development Team met today, we reviewed the Percy usage (screenshot below).  We were running 1000+ Percy snapshots per day, though the developers confirmed they were not looking at 1000's of screenshots, on these days or expecting 1000's of screenshots. 

Outcome: Team confirmed that Percy is misconfigured to generate snapshots for foundation.mozilla.org when not needed or expected.

<img width="1124" alt="Percy usage July - Aug 2022" src="https://user-images.githubusercontent.com/7907755/183724621-c6374410-0152-453f-bb29-3e22545af03e.png">

## Percy is still being required before merging PRs to main

@danielfmiranda supplied an active PR https://github.com/mozilla/foundation.mozilla.org/pull/9199 allowing us to see that merging to main and closing PR's were blocked because Percy was failing (since we have used up all 5000 free screenshots.) 

## Disabled Percy branch protection 

We updated the branch protection settings of the foundation.mozilla.org repo, removing a line that said "percy/foundation.mozilla.org" 

@danielfmiranda then pushed an empty commit to the blocked PR and the Percy checks were removed and the merging was unblocked. 

# Outcome: Percy checks no longer required 
The outcome is that Percy checks are no longer required, for this foundation.mozilla.org repo, before merging to main

**Comment by mofodevops** [comment]
## Investigate why Percy is running 1000+ times a day

Prior to changes on Aug 9, 2022 (described above) Percy was setup to run as a [Github "branch protection rule" for the branch "main."](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#about-branch-protection-rules)    

So any PR that pushed to main (which is most PR's) would cause Percy to run. Even dependabot PRs.  Each time a PR was merged into the "main" branch (thereby updating the "main" branch), all the unmerged PR's would re-run the Github "branch protection rule" to compare to the updated "main" branch.  This is what is generating 1000s of Percy runs per day.  

## Percy Free Tier screenshot allowance = 5000 / month
_**Currently as of Aug 9**_, Free tier of Percy provides 5000 screenshots per month _(subject to change without notification from Percy / Browserstack)_  

For up-to-date screenshot allowances for Percy's Free Tier, see [Browserstack -> Pricing -> Percy](https://www.browserstack.com/accounts/subscriptions?product=percy) 

We have maxed out our Percy screenshots until at least Aug 28, 2022. 

# Create a process to selectively run Percy. 
The ideal solution would be to run Percy only when needed for specific PR's.
This is an area of future work.

**Comment by tbrlpld** [comment]
Thanks for this investigation @mofodevops 

Seems like the amount of open PRs (mostly Dependabot) is at least part of the issue / cost here. I guess this should be another pointer for us to make it a priority to put a workflow in place to deal with them. 

cc @mtdenton
